langchain_openai_api_bridge/chat_completion/langchain_stream_adapter.py

```diff
@@ -22,20 +21,29 @@ async def ato_chat_completion_chunk_stream(
         self,
         astream_event: AsyncIterator[StreamEvent],
         id: str = "",
+        custom_event_handler = lambda event: None,
     ) -> AsyncIterator[OpenAIChatCompletionChunkObject]:
         if id == "":
             id = str(uuid.uuid4())
         async for event in astream_event:
-            kind = event["event"]
-            match kind:
-                case "on_chat_model_stream":
-                    chunk = to_openai_chat_completion_chunk_object(
-                        event=event,
-                        id=id,
-                        model=self.llm_model,
-                        system_fingerprint=self.system_fingerprint,
-                    )
-                    yield chunk
+            custom_event = custom_event_handler(event)
+            if custom_event is not None:
+                yield to_openai_chat_completion_chunk_object(
+                    event=custom_event,
+                    id=id,
+                    model=self.llm_model,
+                    system_fingerprint=self.system_fingerprint,
+                )
+            else:
+                kind = event["event"]
+                match kind:
+                    case "on_chat_model_stream":
+                        yield to_openai_chat_completion_chunk_object(
+                            event=event,
+                            id=id,
+                            model=self.llm_model,
+                            system_fingerprint=self.system_fingerprint,
+                        )
```

Comment:
Those 2 functions do not need to be repeated. This is violating the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). 

`event` should be in a variable on top  instead of re-writing the entire `yield to_openai_chat_completion_chunk_object( ...`